tests/test_scripts.py

```diff
@@ -31,6 +31,31 @@ def get_rule_path():
     return os.path.join(get_rules_path(), "lib", "allocate-memory.yml")
 
 
+@pytest.mark.parametrize(
+    # test detect_duplicate_features.py script for 3 cases
+    # - New rule isn't a valid rule
+    # - New rule file name doesn't ends with .yml
+    # - Checks if number of overlaps found for a test rule are correct
+    "overlap_script,args,expected",
+    [
+        pytest.param("detect_duplicate_features.py", [get_rules_path(), "rules/.github/workflows/sync.yml"], 1),
+        pytest.param("detect_duplicate_features.py", [get_rules_path(), "rules/.github/workflows/sync.ym"], 1),
+        pytest.param(
+            "detect_duplicate_features.py",
+            [
+                get_rules_path(),
+                os.path.join(get_rules_path(), "collection/credit-card/parse-credit-card-information.yml"),
+            ],
+            49,
```

Comment:
I'm afraid this could change when we update rules, so I'd say we better create a test rule set and rule like we do in the test_rules and test that here. We don't have to parametrize it and can just add it verbatim inline.

Reply:
I imagined to create a RuleSet like in `test_lib_rules`:
```
rules = capa.rules.RuleSet(
        [
            capa.rules.Rule.from_yaml(
                textwrap.dedent(
                    """
                    rule:
                        meta:
                            name: a lib rule
                            lib: true
                        features:
                            - api: CreateFileA
                    """
                )
            ),
            capa.rules.Rule.from_yaml(
                textwrap.dedent(
                    """
                    rule:
                        meta:
                            name: a standard rule
                            lib: false
                        features:
                            - api: CreateFileW
                    """
                )
            ),
        ]
    )
```

And then a Rule analogously with overlapping features. We can then test the overlaps counts.

Reply:
The script for overlapping requires path to rules instead of Rule/RuleSet class. Should I modify the functions in script to accept Rule/RuleSet too.

Reply:
Either that or we can create temporary files with pytest fixtures like described here https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html.

Reply:
`test_main_single_rule` has an example of this. It should be sufficient to show the overlap between 2 or 3 rules work.

Reply:
Yeah, It look great. We can create a function that creates temporaryfiles and returns their path. These paths can we passed to script as in `test_main_single_rule`. This rule function can be added to `test_scripts.py`.

Reply:
ideally, we test a few like `api`, `number`, and `string`